src/web-components/SelectHandle/index.ts

```diff
@@ -0,0 +1,185 @@
+ * 
+ * ### Slots:
+ * - **`slottedSearch`**: Slot for the search input.
+ * - **`slottedButtons`**: Slot for action buttons.
+ * - **`slottedLoader`**: Slot for a loading spinner or indicator.
+ * 
+ * ### Properties:
+ * - **`handleData`**:  
+ *   The data for the handles expected in the format [
+ *      { name: 'Handle 1', image: 'ipfs://exampleImage1' },
+ *      { name: 'Handle 2', image: 'ipfs://exampleImage2' },
+ *   ].
+ * 
+ * - **`onFirstUpdated`**:  
+ *   A function called during the `firstUpdated()` lifecycle method.
+ * 
+ * - **`onScroll`**:  
+ *   A function attached to the scroller that detects when the user has scrolled.
+ * 
+ * - **`slottedButtonsStyling`**:  
+ *   A string used to apply custom CSS styling to the slotted buttons.
+ * 
+ * - **`slottedSearchStyling`**:  
+ *   A string used to style the slotted search input.
+ * 
+ * - **`onSelectHandle`**:  
+ *   A function that processes the selected handle. It receives the handle object and sends it back to the parent component.
+ * 
+ * - **`loadingImg`**:  
+ *   A boolean flag indicating whether to display a loading spinner on the selected handle. The loading spinner content must be provided through the `slottedLoader` slot.
+ * 
+ * - **`imageUrl`**:
+ *   The URL of the image to display for the selected handle.
+ * 
+ * - **`handle`**:
+ *   The currently selected handle object. expected in the format { name: 'Handle 1', image: 'ipfs://exampleImage1' }.
+ * 
+ * 
+ * ### Example Method:
+ * ```js
+ * function onSelectHandle(handle) {
+ *   this.loadingImg = true;
+ *   console.log('Selected handle:', handle);
+ *   this.loadingImg = false;
+ *   this.handle = handle;
+ * }
+ * 
+ * onScroll: (event: Event) => {
+ *   const mainSection = event.currentTarget as HTMLElement;
+ *   const scrollPosition = mainSection.scrollTop + mainSection.clientHeight;
+ *   const scrollHeight = mainSection.scrollHeight;
+ * 
+ *   // Check if scrolled to bottom (or near bottom)
+ *   if (scrollPosition >= scrollHeight - 50) {
+ *     console.log('Scrolled to bottom');
+ *   }
+ * },
+ * 
+ * ```
+ * 
+ * ### Example Usage:
+ * ```html
+ * <select-handle
+ *   .handleData=${handleData}
+ *   .loadingImg=${loadingImg}
+ *   .imageUrl=${imageUrl}
+ *   .handle=${handle}
+ *   .onFirstUpdated=${onFirstUpdated}
+ *   .onScroll=${onScroll}
+ *   .onSelectHandle=${onSelectHandle}
+ *   .slottedButtonsStyling=${'display: flex;'}>
+ *      <div slot="slottedSearch">${search}</div>
+ *      <div slot="slottedButtons">${buttons}</div>
+ * </select-handle>
+ * ```
+ */
+
+@customElement('select-handle')
+export class SelectHandle extends LitElement {
+    @property({ type: Array }) handleData: any[] = [];
+    @property({ type: Boolean }) loadingImg = false;
+    @property({ type: String }) slottedButtonsStyling: string;
+    @property({ type: String }) slottedSearchStyling: string;
+    @property({ type: String }) imageUrl: string;
+    @property({ type: Object }) handle: any = {};
```

Comment:
I assume this is the active handle? If so, let's name this to `activeHandle` for clarity